docs/MV3.md

```diff
@@ -8,11 +8,34 @@ This document describes the migration process from MV2 to MV3. MV3 is a new vers
 
 The most important change that lead to this migration was how url request interception model was changed. In MV2, we used `webRequest.onBeforeRequest` to intercept requests and redirect them to the local gateway. That process looked something like:
 
-![MV2 Working Diagram](assets/mv2-working.png)
+```mermaid
+flowchart TD
+    A[Web Request] --> B[Companion Blocks]
+    B --> C[Check if request\ncan be served\nover IPFS?]
+    C --> D[Address Contains CID?]
+    D --> |yes| F
+    D --> |no| E[Address Resolves\nOver IPNS?]
+    E --> |yes| F[Redirect to Local\nIPFS Node]
+    E --> |no| G[Continue Request\nNormally]
```

Comment:
it might be useful to clarify which of these blocks is companion logic, and which is the browser.

```
    A[Web Request] --> B[Companion Blocks]:::companion
    B --> C[Check if request\ncan be served\nover IPFS?]:::companion
    C --> D[Address Contains CID?]:::companion
    D --> |yes| F:::companion
    D --> |no| E[Address Resolves\nOver IPNS?]:::companion
    E --> |yes| F[Redirect to Local\nIPFS Node]:::companion
    E --> |no| G[Continue Request\nNormally]:::browser
    classDef companion fill:#0aca9f,stroke:#7f8491,stroke-width:2px;
    classDef browser fill:#d9dbe2,stroke:#7f8491,stroke-width:1px;
```

### Output

colors come from https://github.com/ipfs-shipyard/ipfs-css/blob/main/ipfs.css#L294-L314, using gray-muted and green-muted, but color choice is up to you.

![image](https://github.com/ipfs/ipfs-companion/assets/1173416/b73f4dd4-5f7b-4e77-a282-e29d26d5bb53)